src/editor/codemirror/linting/worker-transport.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  IJSONRPCData,
+  JSONRPCRequestData,
+} from "@open-rpc/client-js/build/Request";
+import { Transport } from "@open-rpc/client-js/build/transports/Transport";
+
+/**
+ * Transport via a Web Worker postMessage.
+ */
+export class WorkerTransport extends Transport {
+  private listener = (e: MessageEvent) => {
+    this.transportRequestManager.resolveResponse(JSON.stringify(e.data));
+  };
+
+  constructor(private worker: Worker) {
+    super();
+  }
+
+  async connect(): Promise<any> {
+    this.worker.addEventListener("message", this.listener);
+  }
+
+  close(): void {
+    this.worker.removeEventListener("message", this.listener);
+  }
+
+  async sendData(
+    data: JSONRPCRequestData,
+    timeout?: number | null
+  ): Promise<any> {
+    const promise = this.transportRequestManager.addRequest(data, null);
+    this.worker.postMessage((data as IJSONRPCData).request);
+    return promise;
```

Comment:
I'd like to convince myself this is correct. There's lots of code in the language server client that doesn't await promises and I don't understand why.

Reply:
Investigated this a bit and following the flow of the code by hand I don't see how the rpc library doesn't leak a promise for every notification it sends. Needs proper debugging.

Reply:
It does leak. The design requires each transport to settle the promises for notifications, which isn't done in the postMessage based ones. I might be missing a reason why but as notifications are a requests without a response I think it's probably a bug.

Raised https://github.com/open-rpc/client-js/issues/294

Reply:
This is resolved if we go with #251. It also looks to be fixed upstream but I've not upgraded. There's quite a win from the VS Code JSON-RPC package's type support so I think it's the right route for us.

Reply:
It worked out well.